actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java

```diff
@@ -23,169 +26,295 @@
+        ProjectLogger.log(
+            "CertificateActor:getCertificate: cert details merged to user id : " + mergerId,
+            LoggerEnum.INFO.name());
+      } else {
+        ProjectLogger.log(
+            "CertificateActor:getCertificate: cert details unavailable for user id : " + mergeeId,
+            LoggerEnum.ERROR.name());
+      }
     }
+    userResult.setResponseCode(ResponseCode.success);
+    sender().tell(userResult, self());
+    sendMergeNotification(mergeeId, mergerId);
+  }
 
-    /**
-     * This method validates the access-code of the certificate and retrieve certificate details.
-     *
-     * @param userRequest
-     */
-    private void getCertificate(Request userRequest) throws IOException {
-        Map request = userRequest.getRequest();
-        String certificatedId = (String) request.get(JsonKey.CERT_ID);
-        String accessCode = (String) request.get(JsonKey.ACCESS_CODE);
-        Map<String, Object> responseDetails = getCertificateDetails(certificatedId);
-        if (responseDetails.get(JsonKey.ACCESS_CODE.toLowerCase()).equals(accessCode)) {
-            Map userResponse = new HashMap<String, Object>();
-            Response userResult = new Response();
-            Map recordStore = (Map<String, Object>) responseDetails.get(JsonKey.STORE);
-            String jsonData = (String) recordStore.get(JsonKey.JSON_DATA);
-            userResponse.put(JsonKey.JSON, objectMapper.readValue(jsonData,Map.class));
-            userResponse.put(JsonKey.PDF, recordStore.get(JsonKey.PDF_URL));
-            userResponse.put(JsonKey.COURSE_ID,recordStore.get(JsonKey.COURSE_ID));
-            userResponse.put(JsonKey.BATCH_ID,recordStore.get(JsonKey.BATCH_ID));
-            ProjectLogger.log(
-                    "CertificateActor:getCertificate: userMap got with certificateId ".concat(certificatedId+"")+" and response got "+userResponse,
-                    LoggerEnum.INFO.name());
-            userResult.put(JsonKey.RESPONSE, userResponse);
-            sender().tell(userResult, self());
-        } else {
-            ProjectLogger.log(
-                    "CertificateActor:getCertificate: access code is incorrect : " + accessCode,
-                    LoggerEnum.ERROR.name());
-            throw new ProjectCommonException(
-                    ResponseCode.invalidParameter.getErrorCode(),
-                    ProjectUtil.formatMessage(
-                            ResponseCode.invalidParameter.getErrorMessage(), JsonKey.ACCESS_CODE),
-                    ResponseCode.CLIENT_ERROR.getResponseCode());
-        }
+  private void sendMergeNotification(String mergeeId, String mergerId) {
+    ProjectLogger.log(
+        "CertificateActor:sendMergeNotification start sending merge notification to user "
+            + mergeeId
+            + " -"
+            + mergerId,
+        LoggerEnum.INFO.name());
+    HashMap<String, Object> mergeeUserMap = getUserByIdentifier(mergeeId);
```

Comment:
Instead of calling cassandra two times its better to use 
Response getRecordsByIdsWithSpecifiedColumns(
      String keyspaceName, String tableName, List<String> properties, List<String> ids);
below method from CassandraOperation interface